web/src/localization/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { mapBundleSync } from '@fluent/sequence';
+import { negotiateLanguages } from '@fluent/langneg';
+import { Accessor, JSX, createContext, createMemo, useContext } from 'solid-js';
+import { createLocalStorageSignal } from '../utils/localstorage';
+import { Option } from '../types';
+
+const ftls = import.meta.glob('./locale/*.ftl', { eager: true, as: 'raw' });
+const bundles = Object.fromEntries(
+  Object.entries(ftls).map(([fileName, ftl]) => {
+    const languageName = fileName.replace(
+      /^.*\/(.+)\..*$/,
+      (_, language: string) => language,
+    );
+    const resource = new FluentResource(ftl);
+    const bundle = new FluentBundle(languageName);
+    const errors = bundle.addResource(resource);
+    if (errors.length > 0) {
+      throw new Error(
+        `Failed to load translation bundle ${fileName} as ${languageName}`,
+        { cause: errors[0] },
+      );
+    }
+    return [languageName, bundle];
+  }),
+);
+
+const fallbackLanguage = 'en';
+const LanguageContext = createContext<
+  [
+    Accessor<string[]>,
+    {
+      setUserLanguage: (language: Option<string>) => void;
+      availableLanguages: () => { [language: string]: FluentBundle };
+    },
+  ]
```

Comment:
This type can be extracted instead of defining this inline.

Reply:
c4271ca60d171477b8f16eccd278cddd42ecd5ad